src/services/item/plugins/publication/validation/validationQueue.ts

```diff
@@ -0,0 +1,87 @@
+import { hoursToSeconds } from 'date-fns';
+import { Redis } from 'ioredis';
+import { singleton } from 'tsyringe';
+
+import { getIdsFromPath } from '@graasp/sdk';
+
+import { ItemValidationAlreadyExist } from './errors';
+
+const EXPIRES_IN_SECONDS = hoursToSeconds(2);
+const REDIS_PROGRESS_VALIDATION_KEY = 'validation_in_progress';
+const buildRedisProgressKey = (itemId: string) => `${REDIS_PROGRESS_VALIDATION_KEY}:${itemId}`;
+
+/**
+ * Returns the item id of the last item in the path.
+ * @param itemPath The item path to extract the id.
+ * @returns the id of the last item in the path.
+ */
+const getItemIdFromPath = (itemPath: string) => {
+  const itemIds = getIdsFromPath(itemPath);
+  const len = itemIds.length;
+  return itemIds[len - 1];
+};
```

Comment:
I think we have sth like this in sdk already?  `getChildFromPath`

Reply:
Oh yeah, I didn't found it, thanks !